indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -443,8 +443,14 @@ private void startPendingTaskOnRunner(TaskEntry entry, ListenableFuture<TaskStat
                   e.getMessage()
               );
             }
-            TaskStatus taskStatus = TaskStatus.failure(task.getId(), errorMessage);
-            notifyStatus(entry, taskStatus, taskStatus.getErrorMsg());
+            try {
+              TaskStatus taskStatus = TaskStatus.failure(task.getId(), errorMessage);
+              notifyStatus(entry, taskStatus, taskStatus.getErrorMsg());
+              emitTaskCompletionLogsAndMetrics(task, taskStatus);
+            }
+            catch (Exception e2) {
```

Comment:
Can we avoid the try-catch here? Under what conditions can these lines throw an exception?